pkg/idtools/idtools_unix.go

```diff
@@ -229,3 +229,48 @@ func lazyChown(p string, uid, gid int, stat *system.StatT) error {
 	}
 	return os.Chown(p, uid, gid)
 }
+
+// NewIdentityMapping takes a requested username and
+// using the data from /etc/sub{uid,gid} ranges, creates the
+// proper uid and gid remapping ranges for that user/group pair
+func NewIdentityMapping(username string) (*IdentityMapping, error) {
+	usr, err := LookupUser(username)
+	if err != nil {
+		return nil, fmt.Errorf("Could not get usr for username %s: %v", username, err)
+	}
+
+	uid := fmt.Sprintf("%d", usr.Uid)
+
+	subuidRangesWithUserName, err := parseSubuid(username)
+	if err != nil {
+		return nil, err
+	}
+	subgidRangesWithUserName, err := parseSubgid(username)
+	if err != nil {
+		return nil, err
+	}
+
+	subuidRangesWithUID, err := parseSubuid(uid)
+	if err != nil {
+		return nil, err
+	}
+	subgidRangesWithUID, err := parseSubgid(uid)
+	if err != nil {
+		return nil, err
+	}
+
+	subuidRangesWithUserName = append(subuidRangesWithUserName, subuidRangesWithUID...)
+	subgidRangesWithUserName = append(subgidRangesWithUserName, subgidRangesWithUID...)
+
+	if len(subuidRangesWithUserName) == 0 {
+		return nil, fmt.Errorf("No subuid ranges found for usr %q", username)
```

Comment:
typo: usr

Reply:
This one intentially left as `usr`, because the `user` package is clashing with the variable name.

Reply:
error string, not variable name

Reply:
also the error string should start with a lower character

Reply:
I think @AkihiroSuda means the error (`No subuid ranges found for usr`)

We also might want to consider;

- not capitalise the error (`no subuid ranges found for user %q`)
- perhaps `errors.Errorf()` instead of `fmt.Errorf`

Reply:
Done.